lib/server.js

```diff
@@ -407,14 +407,11 @@ Server.prototype.onWebSocket = function(req, socket){
 Server.prototype.attach = function(server, options){
   var self = this;
   var options = options || {};
-  var path = (options.path || '/engine.io').replace(/\/$/, '');
+  var path = (options.path || '/engine.io/');
 
   var destroyUpgrade = (options.destroyUpgrade !== undefined) ? options.destroyUpgrade : true;
   var destroyUpgradeTimeout = options.destroyUpgradeTimeout || 1000;
 
-  // normalize path
-  path += '/';
-
   function check (req) {
     return path == req.url.substr(0, path.length);
```

Comment:
We'd need to fix `check` function, since it wrongly matches a path having a same prefix.

```
console.log(path); // '"/foo"
console.log(req.url); // "/foobar"
check(req); // true
```